vmm/src/cpu.rs

```diff
@@ -2092,6 +2096,12 @@ impl Pausable for CpuManager {
             }
         }
 
+        for state in self.vcpu_states.iter() {
+            if state.active() {
+                while !state.paused.load(Ordering::SeqCst) {}
```

Comment:
This looks scary - there could be a "priority inversion" issue here - can you do it like this: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/vmm/src/cpu.rs#L572-L575 and make sure you have this comment too.

Reply:
Done :)